src/nvim/getchar.c

```diff
@@ -4220,3 +4221,27 @@ static bool typebuf_match_len(const uint8_t *str, int *mlen)
   *mlen = i;
   return str[i] == NUL;  // matched the whole string
 }
+
+/// Retrieve the mapblock at the index either globally or for a certain buffer
+///
+/// @param  index  The index in the maphash[]
+/// @param  buffer_id  -1 for global, else a particular buffer
+mapblock_T *get_maphash(int index, long buffer_id)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
+{
+  if (index > MAX_MAPHASH) {
+    return NULL;
+  }
+
+  if (buffer_id < 0) {
+    return maphash[index];
+  } else {
+    buf_T *buf = handle_get_buffer((handle_T)(int)buffer_id);
```

Comment:
This cast is incorrect, you could just cast to `handle_T`. Though this call should not be there in any case.